### PR TITLE
[TypeDeclaration] Add StrictArrayParamDimFetchRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: packages/Config/RectorConfig.php [hunk 1]
```diff
@@ -169,6 +169,7 @@ public function rule(string $rectorClass): void
     public function rules(array $rectorClasses): void
     {
         Assert::allString($rectorClasses);
+        Assert::uniqueValues($rectorClasses);
```
Comment (TomasVotruba) [inline packages/Config/RectorConfig.php:172]: Ultimate solution to duplicated in-set rules :)

cc @buffcode 
Ref https://github.com/rectorphp/rector-src/pull/4381

File: rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        return [ClassMethod::class, Function_::class];
+    }
+
+    /**
+     * @param ClassMethod|Function_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $hasChanged = false;
+
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php:66]: Like other rule, change param require check with override guard

https://github.com/rectorphp/rector-src/blob/5a91bde755a1906e932d8346466c98101a31ec02/rules/TypeDeclaration/Rector/ClassMethod/ParamTypeByMethodCallTypeRector.php#L149

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php:66]: I see, could you add it?

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php:66]: I will try 👍

Comment (samsonasik) [reply rules/TypeDeclaration/Rector/ClassMethod/StrictArrayParamDimFetchRector.php:66]: https://github.com/rectorphp/rector-src/pull/4386

**Pull request merged**